test/integration/sockaddr_test.go

```diff
@@ -0,0 +1,22 @@
+//+build integration
+//+build !windows
+
+package integration
+
+import (
+	. "gopkg.in/check.v1"
+
+	"github.com/gotestyourself/gotestyourself/icmd"
+)
+
+type SockaddrSuite struct{}
+
+var _ = Suite(&SockaddrSuite{})
+
+func (s *SockaddrSuite) TestSockaddr(c *C) {
+	result := icmd.RunCommand(GomplateBin, "-i",
+		`{{ range (sockaddr.GetAllInterfaces | sockaddr.Exclude "type" "ipv6") -}}
+{{ . | sockaddr.Attr "address" }}
+{{end}}`)
+	result.Assert(c, icmd.Expected{ExitCode: 0, Out: "127.0.0.1"})
+}
```

Comment:
I need to go back and write a helper function that's builtin to `go-sockaddr` that makes testing easier.

Reply:
the ability to inject some fake interface/address state to `go-sockaddr` would be useful for testing, but not entirely crucial...